fda/basis.py

```diff
@@ -1991,6 +1991,52 @@ def copy(self, *, basis=None, coefficients=None, dataset_label=None,
                           axes_labels=axes_labels, extrapolation=extrapolation,
                           keepdims=keepdims)
 
+    def times(self, other):
+        """"Provides a numerical approximation of the multiplication between
+            an FDataObject to other object
+
+        Args:
+            other (int, list, FDataBasis): Object to multiply with the
+                                           FDataBasis object.
+                - int: Multiplies all samples with the value
+                - list: multiply each values with the samples respectively.
+                        Length should match with FDataBasis samples
+                - FDataBasis: if there is one sample it multiplies this with
+                              all the samples in the object. If not, it
+                              multiplies each sample respectively. Samples
+                              should match
+
+        Returns:
+            (FDataBasis): FDataBasis object containing the multiplication
+
+        """
+        if isinstance(other, FDataBasis):
+
+            if not _same_domain(self.domain_range, other.domain_range):
+                raise ValueError("The functions domains are different.")
+
+            basisobj = self.basis.basis_of_product(other.basis)
+            neval = max(10 * max(self.nbasis, other.nbasis) + 1, 201)
+            (left, right) = self.domain_range[0]
+            evalarg = numpy.linspace(left, right, neval)
+
+            first = self.copy(coefficients=(numpy.repeat(self.coefficients,
```

Comment:
What happens if both have only one sample? Why is that an special case?

Reply:
If both has only one sample it will do well. Because this condition only looks if there is a fdatabasis that only has one sample it copies it to the samples of the other object. If both has only one sample, it will do nothing.